rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/aws/lambda/TbLambdaNode.java

```diff
@@ -0,0 +1,183 @@
+                    .withClientConfiguration(new ClientConfiguration()
+                            .withConnectionTimeout(10000)
+                            .withRequestTimeout(5000))
+                    .build();
+        } catch (Exception e) {
+            throw new TbNodeException(e, true);
+        }
+    }
+
+    @Override
+    public void onMsg(TbContext ctx, TbMsg msg) throws ExecutionException, InterruptedException, TbNodeException {
+        var tbMsg = ackIfNeeded(ctx, msg);
+        ObjectNode requestBody = getRequestBody(tbMsg);
+        InvokeRequest request = toRequest(requestBody, this.config.getFunctionName());
+        this.client.invokeAsync(request, new AsyncHandler<>() {
+            @Override
+            public void onError(Exception e) {
+                tellFailure(ctx, tbMsg, e);
+            }
+            @Override
+            public void onSuccess(InvokeRequest request, InvokeResult invokeResult) {
+                try {
+                    if (invokeResult.getFunctionError() != null) {
+                        String errorMessage = invokeResult.getPayload() != null ?
+                                JacksonUtil.toString(getPayload(invokeResult)) : invokeResult.getFunctionError();
+                        throw new RuntimeException(errorMessage);
+                    }
+                    tellSuccess(ctx, getResponseMsg(tbMsg, invokeResult));
+                } catch (Exception e) {
+                    tellFailure(ctx, processException(tbMsg, e), e);
+                }
+            }
+        });
+    }
+
+    private void checkIfInputKeysAreNotEmptyOrElseThrow(Map<String, String> inputKeys) throws TbNodeException {
+        if (inputKeys == null || inputKeys.isEmpty()) {
+            throw new TbNodeException("At least one input key should be specified!", true);
+        }
+    }
+
+    private ObjectNode getRequestBody(TbMsg msg) {
+        ObjectNode requestBodyJson = JacksonUtil.newObjectNode();
+        Map<String, String> mappings = processInputKeys(msg);
+        for (Map.Entry<String, String> entry : mappings.entrySet()) {
+            requestBodyJson.put(entry.getKey(), entry.getValue());
+        }
+        return requestBodyJson;
+    }
+
+    private Map<String, String> processInputKeys(TbMsg msg) {
+        JsonNode msgData = JacksonUtil.toJsonNode(msg.getData());
+        var mappings = new HashMap<String, String>();
+        this.config.getInputKeys().forEach((funcKey, msgKey) -> {
+            String patternProcessedFuncKey = TbNodeUtils.processPattern(funcKey, msg);
+            String patternProcessedMsgValue;
+            try {
+                String patternProcessedMsgKey = TbNodeUtils.processPattern(msgKey, msg);
+                patternProcessedMsgValue = msgData.get(patternProcessedMsgKey).asText();
+            } catch (Exception e) {
+                patternProcessedMsgValue = msgKey;
+            }
+            mappings.put(patternProcessedFuncKey, patternProcessedMsgValue);
```

Comment:
I think that it's better to replace the string msgKey with something like an argument in the MathNode. Let's start with Message, Metadata. Constant.